### PR TITLE
#1992 fix: Remove currentUserEmail from GoogleAuthManager

### LINKED ISSUE #1992: Remove currentUserEmail from GoogleAuthManager
`GoogleAuthManager` needs to be refactored to not require currentUserEmail

https://github.com/FlowCrypt/flowcrypt-ios/blob/master/FlowCrypt/Functionality/Services/Google%20User%20Service/GoogleAuthManager.swift#L54

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR removed currentUserEmail from GoogleAuthManager

close #1992 // if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Functionality/Api/IdTokenUtils.swift [hunk 1]
```diff
@@ -16,10 +16,9 @@ class IdTokenUtils {
         }
 
         let googleAuthManager = GoogleAuthManager(
-            currentUserEmail: userEmail ?? user?.email,
             appDelegateGoogleSessionContainer: nil
         )
 
-        return try await googleAuthManager.getCachedOrRefreshedIdToken()
+        return try await googleAuthManager.getCachedOrRefreshedIdToken(email: userEmail)
```
Comment (sosnovsky) [inline FlowCrypt/Functionality/Api/IdTokenUtils.swift:22]: Probably it should be 
```swift
googleAuthManager.getCachedOrRefreshedIdToken(email: userEmail ?? user?.email)
```

File: FlowCrypt/Functionality/Api/IdTokenUtils.swift [hunk 2]
```diff
@@ -16,10 +16,9 @@ class IdTokenUtils {
         }
 
         let googleAuthManager = GoogleAuthManager(
-            currentUserEmail: userEmail ?? user?.email,
             appDelegateGoogleSessionContainer: nil
```
Comment (sosnovsky) [inline FlowCrypt/Functionality/Api/IdTokenUtils.swift:19]: `appDelegateGoogleSessionContainer` has default value `nil` in `GoogleAuthManager` initializer, so we can remove it here

File: FlowCrypt/Functionality/Services/Google User Service/GoogleAuthManager.swift [hunk 3]
```diff
@@ -216,20 +212,12 @@ extension GoogleAuthManager {
 
     // save auth session to keychain
     private func saveAuth(state: OIDAuthState, for email: String) {
+        currentUserEmail = email // Save email variable so that OIDAuthStateChangeDelegate can access
```
Comment (sosnovsky) [inline FlowCrypt/Functionality/Services/Google User Service/GoogleAuthManager.swift:215]: I think we can get `email` value from `OIDAuthState`, so no need to separate `email` param:
```swift
let authorization = GTMAppAuthFetcherAuthorization(authState: state)
GTMAppAuthFetcherAuthorization.save(authorization, toKeychainForName: Constants.index + authorization.userEmail)
```
What do you think, will it work here?

Comment (ioanmo226) [reply FlowCrypt/Functionality/Services/Google User Service/GoogleAuthManager.swift:215]: Yeah, You are right.
I think it will work.

**Review by sosnovsky — APPROVED** [review]
👍

**Pull request merged**